src/ogr/geometry.jl

```diff
@@ -940,19 +956,143 @@ function empty!(geom::G)::G where {G<:AbstractGeometry}
+    wkbCompoundCurveZM,
+    wkbCurvePolygonZM,
+    wkbMultiCurveZM,
+    wkbMultiSurfaceZM,
+    wkbCurveZM,
+    wkbSurfaceZM,
+    wkbPolyhedralSurfaceZM,
+    wkbTINZM,
+    wkbTriangleZM,
+)
+const wkbEnums25D = (
+    wkbPoint25D,
+    wkbLineString25D,
+    wkbPolygon25D,
+    wkbMultiPoint25D,
+    wkbMultiLineString25D,
+    wkbMultiPolygon25D,
+    wkbGeometryCollection25D,
+)
+
+const wkbEnums2d = (wkbEnums..., wkbEnumsM...)
+const wkbEnums3d = (wkbEnumsZ..., wkbEnumsZM..., wkbEnums25D...)
+
+const _AbstractGeometry = Union{map(x -> AbstractGeometry{x}, wkbEnums)...}
+const _IGeometry = Union{map(x -> IGeometry{x}, wkbEnums)...}
+const _Geometry = Union{map(x -> Geometry{x}, wkbEnums)...}
+
+const _AbstractGeometryM = Union{map(x -> AbstractGeometry{x}, wkbEnumsM)...}
+const _IGeometryM = Union{map(x -> IGeometry{x}, wkbEnumsM)...}
+const _GeometryM = Union{map(x -> Geometry{x}, wkbEnumsM)...}
+
+const _AbstractGeometryZ = Union{map(x -> AbstractGeometry{x}, wkbEnumsZ)...}
+const _IGeometryZ = Union{map(x -> IGeometry{x}, wkbEnumsZ)...}
+const _GeometryZ = Union{map(x -> Geometry{x}, wkbEnumsZ)...}
+
+const _AbstractGeometry25D = Union{map(x -> AbstractGeometry{x}, wkbEnums25D)...}
+const _IGeometry25D = Union{map(x -> IGeometry{x}, wkbEnums25D)...}
+const _Geometry25D = Union{map(x -> Geometry{x}, wkbEnums25D)...}
+
+const _AbstractGeometryZM = Union{map(x -> AbstractGeometry{x}, wkbEnumsZM)...}
+const _IGeometryZM = Union{map(x -> IGeometry{x}, wkbEnumsZM)...}
+const _GeometryZM = Union{map(x -> Geometry{x}, wkbEnumsZM)...}
+
+const _AbstractGeometry2d = Union{map(x -> AbstractGeometry{x}, wkbEnums2d)...}
+const _IGeometry2d = Union{map(x -> IGeometry{x}, wkbEnums2d)...}
+const _Geometry2d = Union{map(x -> Geometry{x}, wkbEnums2d)...}
+
+const _AbstractGeometry3d = Union{map(x -> AbstractGeometry{x}, wkbEnums3d)...}
+const _IGeometry3d = Union{map(x -> IGeometry{x}, wkbEnums3d)...}
+const _Geometry3d = Union{map(x -> Geometry{x}, wkbEnums3d)...}
+
+const _AbstractGeometryHasM = Union{_AbstractGeometryM,_AbstractGeometryZM}
+const _IGeometryHasM = Union{_IGeometryM,_IGeometryZM}
+const _GeometryHasM = Union{_GeometryM,_GeometryZM}
+
+const _AbstractGeometryNoM = Union{_AbstractGeometry,_AbstractGeometryZ,_AbstractGeometry25D}
+const _IGeometryNoM = Union{_IGeometry,_IGeometryZ,_IGeometry25D}
+const _GeometryNoM = Union{_Geometry,_GeometryZ,_Geometry25D}
+
 """
     is3d(geom::AbstractGeometry)
 
 Returns `true` if the geometry has a z coordinate, otherwise `false`.
 """
-is3d(geom::AbstractGeometry)::Bool = GDAL.ogr_g_is3d(geom.ptr) != 0
+is3d(geom::AbstractGeometry)::Bool = GDAL.ogr_g_is3d(geom.ptr) != 0 # Is a fallback still needed?
+is3d(geom::_AbstractGeometry2d) = false
+is3d(geom::_AbstractGeometry3d) = true
 
 """
     ismeasured(geom::AbstractGeometry)
 
 Returns `true` if the geometry has a m coordinate, otherwise `false`.
 """
 ismeasured(geom::AbstractGeometry)::Bool = GDAL.ogr_g_ismeasured(geom.ptr) != 0
```

Comment:
Can we add a unit test for this line? It seems the current unit tests only covers the new lines.

Reply:
I've deleted it, it's never hit now.